File: frontend/src/api/customer.ts
import request from './request'

export interface CustomerListItem {
  id: string
  customer_no: string
  company_name: string
  company_short_name: string | null
  industry: string | null
  company_size: string | null
  province: string | null
  city: string | null
  customer_level: string | null
  status: string
  assigned_sales: string | null
  created_at: string
}

export interface Contact {
  id: string
  customer_id: string
  name: string
  title: string | null
  department: string | null
  phone: string
  email: string | null
  wechat: string | null
  is_primary: boolean
  created_at: string
}

export interface CustomerDetail extends CustomerListItem {
  unified_social_credit_code: string | null
  legal_representative: string | null
  address: string | null
  remark: string | null
  contacts: Contact[]
}

export interface FollowUp {
  id: string
  customer_id: string
  follow_type: string
  content: string
  next_date: string | null
  created_by: string | null
  created_at: string
}

export const customerApi = {
  list(params: { page?: number; page_size?: number; status?: string; keyword?: string }) {
    return request.get<any, { code: number; data: CustomerListItem[]; total: number }>('/customers', { params })
  },
  get(id: string) {
    return request.get<any, { code: number; data: CustomerDetail }>(`/customers/${id}`)
  },
  create(data: Record<string, unknown>) {
    return request.post<any, { code: number; data: CustomerDetail }>('/customers', data)
  },
  update(id: string, data: Record<string, unknown>) {
    return request.put<any, { code: number; data: CustomerDetail }>(`/customers/${id}`, data)
  },
  delete(id: string) {
    return request.delete(`/customers/${id}`)
  },
  addContact(customerId: string, data: Record<string, unknown>) {
    return request.post(`/customers/${customerId}/contacts`, data)
  },
  updateContact(customerId: string, contactId: string, data: Record<string, unknown>) {
    return request.put(`/customers/${customerId}/contacts/${contactId}`, data)
  },
  deleteContact(customerId: string, contactId: string) {
    return request.delete(`/customers/${customerId}/contacts/${contactId}`)
  },
  listFollowUps(customerId: string) {
    return request.get<any, { code: number; data: FollowUp[] }>(`/customers/${customerId}/follow-ups`)
  },
  addFollowUp(customerId: string, data: Record<string, unknown>) {
    return request.post(`/customers/${customerId}/follow-ups`, data)
  },
}

File: backend/app/schemas/customer.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


# ── 联系人 ──────────────────────────────────────────────────
class ContactBase(BaseModel):
    name: str
    title: str | None = None
    department: str | None = None
    phone: str
    email: str | None = None
    wechat: str | None = None
    is_primary: bool = False


class ContactCreate(ContactBase):
    pass


class ContactUpdate(BaseModel):
    name: str | None = None
    title: str | None = None
    department: str | None = None
    phone: str | None = None
    email: str | None = None
    wechat: str | None = None
    is_primary: bool | None = None


class ContactOut(ContactBase):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    customer_id: uuid.UUID
    created_at: datetime


# ── 跟进记录 ────────────────────────────────────────────────
class FollowUpCreate(BaseModel):
    follow_type: str
    content: str
    next_date: str | None = None


class FollowUpOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    customer_id: uuid.UUID
    follow_type: str
    content: str
    next_date: str | None
    created_by: uuid.UUID | None
    created_at: datetime


# ── 客户 ────────────────────────────────────────────────────
class CustomerBase(BaseModel):
    company_name: str
    company_short_name: str | None = None
    unified_social_credit_code: str | None = None
    legal_representative: str | None = None
    industry: str | None = None
    company_size: str | None = None
    province: str | None = None
    city: str | None = None
    address: str | None = None
    customer_level: str | None = None
    status: str = "潜在"
    assigned_sales: uuid.UUID | None = None
    remark: str | None = None


class CustomerCreate(CustomerBase):
    contacts: list[ContactCreate] = []


class CustomerUpdate(BaseModel):
    company_name: str | None = None
    company_short_name: str | None = None
    unified_social_credit_code: str | None = None
    legal_representative: str | None = None
    industry: str | None = None
    company_size: str | None = None
    province: str | None = None
    city: str | None = None
    address: str | None = None
    customer_level: str | None = None
    status: str | None = None
    assigned_sales: uuid.UUID | None = None
    remark: str | None = None
    contacts: list[ContactCreate] | None = None


class CustomerOut(CustomerBase):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    customer_no: str
    created_at: datetime
    contacts: list[ContactOut] = []


class CustomerListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    customer_no: str
    company_name: str
    company_short_name: str | None
    industry: str | None
    company_size: str | None
    province: str | None
    city: str | None
    customer_level: str | None
    status: str
    assigned_sales: uuid.UUID | None
    created_at: datetime
